src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/service/PrisonerSearchService.kt

```diff
@@ -39,15 +40,30 @@ class PrisonerSearchService(
       body.getSortBy(),
     )
 
-    val resultsWithLicences: List<Pair<CaseloadResult, Licence?>> =
+    val deliusRecordsWithLicences: List<Pair<CaseloadResult, Licence?>> =
       teamCaseloadResult.map { it to getLicence(it) }
 
-    val prisonerRecords = findPrisonersForRelevantRecords(resultsWithLicences)
+    val prisonerRecords = findPrisonersForRelevantRecords(deliusRecordsWithLicences)
 
-    val searchResults = resultsWithLicences.mapNotNull { (result, licence) ->
-      when (licence) {
-        null -> createNotStartedRecord(result, prisonerRecords[result.identifiers.noms])
-        else -> createRecord(result, licence, prisonerRecords[result.identifiers.noms])
+    val cvlSearchRecords = deliusRecordsWithLicences.map { (caseloadResult, licence) ->
+      val prisonRecord = prisonerRecords[caseloadResult.identifiers.noms]
+      CvlProbationSearchRecord(
+        caseloadResult = caseloadResult,
+        prisonerSearchPrisoner = prisonRecord,
+        licence = licence,
+      )
+    }
+
+    val prisonersWithoutLicences = cvlSearchRecords.filter { searchRecord -> searchRecord.licence == null }.map {
+      it.prisonerSearchPrisoner
+    }
+    val licenceStartDates = releaseDateService.getLicenceStartDates(prisonersWithoutLicences)
+
+    val searchResults = cvlSearchRecords.mapNotNull {
+      val licenceStartDate = licenceStartDates[it.prisonerSearchPrisoner?.prisonerNumber]
```

Comment:
The nullable `prisonerSearchPrisoner` here is throwing me a bit - so we would have a Delius record and no NOMIS record? I always thought the reverse was possible (NOMIS record but no Delius record) but not this way round?

Reply:
There _shouldn't_ ever be a time that there's no NOMIS record, but in reality there can be a disconnect between Delius and NOMIS, whether that's an incomplete link between CRN and noms ID, or a duplicate NOMIS record has been removed and references not been updated.